core/include/thread.h

```diff
@@ -163,6 +162,11 @@ extern "C" {
  */
 typedef void *(*thread_task_func_t)(void *arg);
 
+/**
+ * @brief Forward declaration of msg_t
```

Comment:
I _think_ if you use `@ref struct msg` here, doxygen will correctly render it. All in all, it would be nice to have a link to the actual struct, since this is in a completely different header.